src/outputgeneration/ParseTreeWriter.js

```diff
@@ -1006,9 +1006,12 @@ export class ParseTreeWriter extends ParseTreeVisitor {
    * @param {NewExpression} tree
    */
   visitNewExpression(tree) {
+    var complexOperand = tree.operand.operand !== void 0;
```

Comment:
The writer should not have logic on it.

We never use undefined for valid properties. The right approach is to check the type.

So, if I parse

```js
new a.b.c()
```

Then the operand is `a.b.c` and you now want to add parens around that? That does not seem correct.

I think we need to look at the AST we generate. Is it correct? I think it is. However, our transformer is not aware of `new`.

```js
new a `b` `c`
```

is transformed into

```js
new a($__0)($__1)
```

when we need to transform it into

```js
new (a($__0)($__1))
```